tox.ini

```diff
@@ -1,24 +1,31 @@
 [tox]
 envlist =
-    py3{11,10,9,8,7},pypy3{8,7}
+    py3{12,11,10,9,8,7}
+    pypy39
     style
     typing
     docs
 skip_missing_interpreters = true
 
 [testenv]
+package = wheel
+wheel_build_env = .pkg
 deps = -r requirements/tests.txt
 commands = pytest -v --tb=short --basetemp={envtmpdir} {posargs}
 
 [testenv:style]
 deps = pre-commit
 skip_install = true
-commands = pre-commit run --all-files --show-diff-on-failure
```

Comment:
Is this intentional (not sure if it is useful, but I'd think it is)

Reply:
Yeah, it's annoying when running locally (where you can use `git diff`), and workflow runs that would trigger this are already associated with a commit that shows a better diff.